src/workspaces/NewWorkspaceModal/NewWorkspaceWizard.tsx

```diff
@@ -0,0 +1,798 @@
+            disabled={activeTab === 'basic' && errors}
+            tooltip={getTooltip()}
+            onClick={handleSecondaryButtonClick}
+          >
+            {Utils.cond([activeTab === 'basic', () => 'Quick create workspace'], () => 'Previous')}
+          </ButtonSecondary>
+          <ButtonPrimary
+            style={{ padding: '0 1rem', marginLeft: '1rem' }}
+            disabled={errors}
+            tooltip={activeTab === 'security' ? Utils.summarizeErrors(errors) : undefined}
+            onClick={handlePrimaryButtonClick}
+          >
+            {Utils.cond([activeTab === 'security', () => 'Create workspace'], () => 'Next')}
+          </ButtonPrimary>
+        </div>
+      </div>
+    );
+
+    return Utils.cond(
+      [loading, () => spinnerOverlay],
+      [
+        hasBillingProjects,
+        () => (
+          <Modal
+            title={Utils.cond(
+              [!!title, () => title],
+              [!!cloneWorkspace && creating, () => 'Cloning workspace'],
+              [creating, () => 'Creating workspace'],
+              [!!cloneWorkspace, () => 'Clone Workspace'],
+              () => 'Create a New Workspace'
+            )}
+            // Hold modal open while waiting for create workspace request.
+            shouldCloseOnOverlayClick={!creating}
+            shouldCloseOnEsc={!creating}
+            showButtons={false}
+            onDismiss={onDismiss}
+            okButton={
+              <ButtonPrimary disabled={errors} tooltip={Utils.summarizeErrors(errors)} onClick={create}>
+                {Utils.cond(
+                  [!!buttonText, () => buttonText],
+                  [!!cloneWorkspace, () => 'Clone Workspace'],
+                  () => 'Create Workspace'
+                )}
+              </ButtonPrimary>
+            }
+            width={620}
+            styles={{ modal: { height: 675 } }}
+          >
+            {creating ? (
+              <CreatingWorkspaceMessage />
+            ) : (
+              <>
+                <TabWizard
+                  aria-label='create workspace wizard'
+                  activeTab={activeTab}
+                  tabNames={['basic', 'sharing', 'security']}
+                  tabsDisabled={[false, activeTab === 'basic' && !!errors, activeTab === 'basic' && !!errors]}
+                  displayNames={{
+                    basic: '1. Basic Information and Billing',
+                    sharing: '2. Sharing',
+                    security: '3. Additional Security Options',
+                  }}
+                  style={{ textTransform: 'none', width: '90%' }}
+                  getHref={() => {}}
+                  getOnClick={(currentTab) => {
+                    setActiveTab(currentTab);
+                  }}
+                >
+                  {null /* nothing to display at the end of the tab bar */}
+                </TabWizard>
+                {activeTab === 'basic' && basicTab}
```

Comment:
If you make Tab be an interface, you could also give it a render method, and then this could be simplified with `_.find((tab) => tab.number === activeTab, tabs).render()` so you don't need a line for each tab?